File: src/ main/ java/ MoveZeroes/MoveZeroes.java

public class MoveZeroes {

	public void moveZeroes(int[] nums) {
        for(int i=0; i<nums.length;i++){
        	//获取为0的下标
        	if(nums[i]==0)
        	{
        		int k = i;
        		for(int j=i+1;j<nums.length;j++){
        			if(nums[j]!=0){
        				int a = nums[j];
        				nums[j] = nums[k];
        				nums[k] = a;
        				k=j;
        			}
        		}
        	}
        }
    }

}

File: src/main/java/MoveZeroes/JuChunChen/MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 *  [1,0,0,3,12]
 *  [1,3,0,0,12]
 *  [1,3,12,0,0]
 * 
 */
var moveZeroes = function(nums) {
      let temp = 0;
  let len = nums.length;
  for(let i = 0; i < len; i++){
      if(nums[i] !== 0){
          nums[temp] = nums[i];
          temp++;
      }
  }
  for(; temp < len; temp++){
      nums[temp] = 0;
  }  
};